h2o-core/src/main/java/water/Scope.java

```diff
@@ -92,4 +89,7 @@ static public void untrack(Iterable<Key<Vec>> keys) {
     HashSet<Key> xkeys = scope._keys.peek();
     for (Key<Vec> key : keys) xkeys.remove(key); // Untrack key
   }
+
+  @Override
```

Comment:
The idea of `Closeable` is to enable the following construct:
```
try (Scope scope = new Scope()) {
   // do something
}
```
Then the scope closes automatically at the end of the try-block (even if there is an exception in the middle).

@vpatryshev Could you please add a couple tests to check that this construct indeed works as intended (you may need to ensure that `_keys` is created inside the constructor). The old method of invocation (Scope.enter() / Scope.exit()) should work as well, to avoid any flame wars. Although it would be OK (I think) to mark those 2 as deprecated.